ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -171,6 +172,20 @@ object WebSocketService {
     ): StreamPredicate[Positive]
 
     def renderCreatedMetadata(p: Positive): Map[String, JsValue]
+
+    /** The first item of the tuple is the request to be issued to the
+      * ActiveContractsService, the second to the TransactionService
+      */
+    def splitRequest(
+        prefix: Option[domain.StartingOffset],
+        request: A,
+    ): (Option[A], Option[A])
```

Comment:
If you think it would behoove your implementation to have *different* types for the first and second `A`s here, then by all means, add a type member (just as I did with `Positive`) to represent one or the other.

Additionally, I think you want `\&/` here, because it appears that `(None, None)` is None-sense.

Reply:
> If you think it would behoove your implementation to have different types for the first and second As here, then by all means, add a type member (just as I did with Positive) to represent one or the other.

Yeah, I was a bit unsure of how to manage different types for the `query` and `fetch` endpoints, probably the type parameter is the way to go, I'll try this today, thanks.

> Additionally, I think you want \&/ here, because it appears that (None, None) is None-sense.

Thanks, I'll have a look at this if I end up needing it.

Reply:
Working on a different approach, reverted this one with https://github.com/digital-asset/daml/pull/9847/commits/0a3f85f25b2b85b9b8112c96bf8b9d27f8de0fe6